delta-lake/delta-33x/src/main/scala/com/nvidia/spark/rapids/delta/delta33x/GpuDelta33xParquetFileFormat.scala

```diff
@@ -273,6 +299,133 @@ case class GpuDelta33xParquetFileFormat(
         None
     }
   }
+}
+
+class DeltaMultiFileReaderFactory(
+    @transient sqlConf: SQLConf,
+    broadcastedConf: Broadcast[SerializableConfiguration],
+    dataSchema: StructType,
+    readDataSchema: StructType,
+    partitionSchema: StructType,
+    filters: Array[Filter],
+    @transient rapidsConf: RapidsConf,
+    metrics: Map[String, GpuMetric],
+    useMetadataRowIndex: Boolean,
+    tablePath: Option[String]
+    ) extends GpuParquetMultiFilePartitionReaderFactory(sqlConf, broadcastedConf,
+      dataSchema, readDataSchema, partitionSchema,
+      filters, rapidsConf, metrics, queryUsesInputFile = true) {
+
+  private val schemaWithIndices = readDataSchema.fields.zipWithIndex
+  def findColumn(name: String): Option[ColumnMetadata] = {
+    val results = schemaWithIndices.filter(_._1.name == name)
+    if (results.length > 1) {
+      throw new IllegalArgumentException(
+        s"There are more than one column with name=`$name` requested in the reader output")
+    }
```

Comment:
There is a syntactic sugar for it 


```suggestion
    require (results.length  <=  1,   
       s"There are more than one column with name=`$name` requested in the reader output")
```
The use would understand this message better if it is phrased 

N columns found for name ..., expected number of columns 1 

or something like that